### PR TITLE
[#1493] Create personalized search results

### LINKED ISSUE #1493: Create personalized search results
**Is your feature request related to a problem? Please describe.**
In the current system design, we don't have a place where we can show a sort of personalized search results.  

**Describe the solution you'd like**
- [x] Create a dedicated "Recommended results" section on the search result page. 
- [x] The current implementation (for all users) should be based on the most popular resources from the most popular category. 
- [x] Some users should see view according to Mockup1 and others according to Mockup2
- [x] Use our split configuration to configure the experiment for each implementation

**Additional context**
Mockup1: https://projects.invisionapp.com/share/WAX4U50BV2G#/screens/416192304_Item_List_Search_-_Recommended_Bar_V2
Mockup2: https://projects.invisionapp.com/share/HRX4PQD8NUQ#/screens/416159204_Item_List_Search_-_Recommended_Box

### PR DESCRIPTION
**Pull request by JanKapala**
This PR closes #1493 issue that include creating two versions of recommended services panels.

### TODO:
- [x] Create a dedicated "Recommended results" section on the search result page.
- [x] The current implementation (for all users) should be based on the most popular resources from the most popular category.
- [x] Some users should see view according to Mockup1 and others according to Mockup2
- [x] Use our split configuration to configure the experiment for each implementation
- [x] Fine-tune CSS (@jarekzet )
- [x] Tests:
  - [x] Add split gem helper for testing
  - [x] SimpleRecommender
  - [x] RecommenderSystemHelper
  - [x] ServiceRecommendations
  - [x] Adapt old services feature specs for new layout with recommendations 
- [x] Remove "Based on your search" string from the recommendation panel
- [x] Add possibility of disabling recommendations 
- [x] Fully handle A/B testing by using proper `ab_finished`  

### How to manually test (for reviewers):
Go to the `admin` -> `AB experiments` and try variants `v1`, `v2` of the `recommendation_panel` experiment and compare appearance of `/services` page with mockups (links to mockups available in the #1493 issue). Try to refresh page few times and try few sessions in incognito.
- If you are logged user you should always see the same variant even if you click recommended service or restart the session.
- If you aren't logged user and you don't store cookies (as in incognito mode, it simulates different users) you should see each of two variants of recommendation 50% probability.

Moreover there is `disabled` variant of experiments that is a control group without recommendations.

### Further work proposal:
- Turn on `RecommenderSystemHelper` and `ServiceRecommendations` specs (temporarily disabled because of unknown bug of `use_ab_test` method in specs)
- Refactor of translations in spec folder (?)

### CODE REVIEW

**Comment by agpul** [comment]
image
I recommend removing this statement - this is not true, especially for unlogged users.

**Comment by JanKapala** [comment]
"Based on your search" statement removed

**Review by agpul — APPROVED** [review]

File: app/services/recommender/simple_recommender.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    recommended_services = get_recommended_services n
+    fill_missing recommended_services, n
+  end
+
+  private
+    def get_recommended_services(n)
+      records2services get_records(n)
+    end
+
+    def get_records(n)
```
Comment (goreck888) [inline app/services/recommender/simple_recommender.rb:14]: Avoid using "get" word in method names.

**Review by goreck888 — CHANGES_REQUESTED** [review]
Just one thing to fix, everything else is fine from my point of view.

**Comment by goreck888** [comment]
Moreover 
./spec/features/categories_spec.rb:62
 ./spec/features/service_spec.rb:162
are failing because recommendations are services and Rspec sees it as additional services.
Just turn off split experiment in tests or include recommendations in these.

**Review by goreck888 — APPROVED** [review]

**Review by wujuu — APPROVED** [review]

**Pull request merged**